src/pages/controlled-form/controlled-form.tsx

```diff
@@ -0,0 +1,239 @@
+        password: data.pass1,
+        country: data.country,
+        isMale: data.isMale === "male" ? true : false,
+        isFemale: data.isMale === "female" ? true : false,
+        isAgree: data.isAgree === "yes" ? true : false,
+      }),
+    );
+    dispatch(setIsFilled(true));
+    dispatch(setIsUncFormUpdate(false));
+    dispatch(setIsContFormUpdate(true));
+    if (!Object.is(data.image, {})) {
+      const image = data.image as Blob[];
+      setDataBase(image[0]);
+    }
+    navigator("/");
+  };
+
+  const submitForm: SubmitHandler<Form> = async (data) => {
+    try {
+      await yupSchemaForHookForm.validate({
+        name: data.name,
+        age: data.age,
+        email: data.email,
+        pass1: data.pass1,
+        pass2: data.pass2,
+        country: data.country,
+        isAgree: data.isAgree,
+        image: data.image === null ? {} : data.image,
+      });
+      submitDataToRedux(data);
+    } catch (e) {
+      setSubmitBtnClass("btn_submit_disabled");
+    }
+  };
+
+  const setDataBase = (img: Blob | null) => {
+    if (img) {
+      reader.readAsDataURL(img);
+      reader.onload = () => {
+        dispatch(setDataBase64(reader.result));
+      };
+    }
+  };
+
+  const getCountriesList = () => {
+    const value = getValues("country").toLocaleLowerCase().trim();
+    if (value !== "") {
+      setCountryList(
+        countries.filter((country) => country.toLowerCase().startsWith(value)),
+      );
+    }
+  };
+
+  const setCountry = (value: string) => {
+    setValue("country", value);
+  };
+
+  const closeCountryList = (
+    e: React.MouseEvent<HTMLDivElement, MouseEvent>,
+  ) => {
+    const target = e.target as HTMLDivElement;
+    if (!target.closest(styles.countries_list)) {
+      setClueIsVisible(false);
+    }
+  };
+
+  return (
+    <div
+      className={styles.wrapper}
+      onClick={(e) => {
+        closeCountryList(e);
+      }}
+    >
+      <form
+        className={styles.form}
+        onSubmit={handleSubmit((data) => submitForm(data as Form))}
+      >
+        <label htmlFor="name">Name</label>
+        <input type="text" id="name" {...register("name")} />
+        <p className={styles.error_message}>{errors.name?.message}</p>
+        <label htmlFor="age">Age</label>
+        <input type="text" id="age" {...register("age")} />
+        <p className={styles.error_message}>{errors.age?.message}</p>
+        <label htmlFor="email">Email</label>
+        <input type="email" id="email" {...register("email")} />
+        <p className={styles.error_message}>{errors.email?.message}</p>
+        <label htmlFor="pass1">Password</label>
+        <input type="password" id="pass1" {...register("pass1")} />
+        <p className={styles.error_message}>{errors.pass1?.message}</p>
+        <label htmlFor="pass2">Repeat password</label>
+        <input type="password" id="pass2" {...register("pass2")} />
+        <p className={styles.error_message}>{errors.pass2?.message}</p>
```

Comment:
const inputFields = [
  { label: "Name", id: "name", ref: nameRef, errorKey: "name" },
  { label: "Age", id: "age", ref: ageRef, errorKey: "age" },
  { label: "Email", id: "email", ref: email, errorKey: "email" },
  { label: "Password", id: "pass1", ref: pass1, errorKey: "pass1" },
  { label: "Repeat password", id: "pass2", ref: pass2, errorKey: "pass2" },
];

const renderedInputs = inputFields.map((field) => (
  <div key={field.id}>
    <label htmlFor={field.id}>{field.label}</label>
    <input type={field.id === "email" ? "email" : "text"} id={field.id} ref={field.ref} />
    <p className={styles.error_message}>{errorMessages[field.errorKey]}</p>
  </div>
));

     <form
        className={styles.form}
        onSubmit={handleSubmit((data) => submitForm(data as Form))}
      > {renderedInputs } ... </form>


I have described an approximate example, but I think it would be nice to follow here **(DRY)do'nt repeat yourself** and use all the advantages of **React**